app/src/main/java/com/github/se/cyrcle/ui/map/MapScreen.kt

```diff
@@ -153,69 +155,44 @@ fun MapScreen(
-            val resizedBitmap = Bitmap.createScaledBitmap(bitmap, 100, 150, false)
-
             // Get parkings in the current view
             parkingViewModel.getParkingsInRect(loadedBottomLeft, loadedTopRight)
-
-            // Create PointAnnotationOptions for each parking
-            var pointAnnotationOptions =
-                listOfParkings.value.map { parking ->
-                  PointAnnotationOptions()
-                      .withPoint(parking.location.center)
-                      .withIconImage(resizedBitmap)
-                }
-
-            // Add annotations to the annotation manager and display them on the map
-            pointAnnotationOptions.forEach { annotationManager.create(it) }
-
             // Add a camera change listener to detect zoom changes
-            val cameraChangeListener = OnCameraChangeListener {
-
-              // Remove the view annotation if the user moves the map
-              if (removeViewAnnotation) {
-                viewAnnotationManager.removeAllViewAnnotations()
-                cancelables.forEach(Cancelable::cancel)
-              }
-
-              state.value = mapView.mapboxMap.cameraState.zoom
-
-              // Get the top right and bottom left coordinates of the current view only when
-              // what the user sees is outside the screen
-              val (currentBottomLeft, currentTopRight) =
-                  getScreenCorners(mapView, useBuffer = false)
-              if (!inBounds(currentBottomLeft, currentTopRight, loadedBottomLeft, loadedTopRight)) {
-                Log.d("MapScreen", "Loading parkings in new view")
-                // Get the buffered coordinates for loading parkings
-
-                val loadedCorners = getScreenCorners(mapView, useBuffer = true)
-                loadedBottomLeft = loadedCorners.first
-                loadedTopRight = loadedCorners.second
+            val cameraCancelable =
+                mapView.mapboxMap.subscribeCameraChanged {
+                  // store the zoom level
+                  zoomState.value = mapView.mapboxMap.cameraState.zoom
+
+                  // Remove the view annotation if the user moves the map
+                  if (removeViewAnnotation) {
+                    viewAnnotationManager.removeAllViewAnnotations()
+                    cancelables.forEach(Cancelable::cancel)
+                  }
 
-                parkingViewModel.getParkingsInRect(loadedBottomLeft, loadedTopRight)
+                  // Get the top right and bottom left coordinates of the current view only when
+                  // what the user sees is outside the screen
+                  val (currentBottomLeft, currentTopRight) =
+                      getScreenCorners(mapView, useBuffer = false)
+                  if (!inBounds(
+                      currentBottomLeft, currentTopRight, loadedBottomLeft, loadedTopRight)) {
+                    Log.d("MapScreen", "Loading parkings in new view")
```

Comment:
Is it intended to stay or is it just for debugging sake ?

Reply:
I don't know either, I didn't put this Log.d(), let's ask @Armetiis

Reply:
Is it really that bad to keep some Logs? I really don't think logs should be systematically deleted. I often find myself writing the same logs over and over because I deleted them before pushing. In this case, given that we still have a problem with loading the parkings with certain map settings, I think it's better to keep this log

Reply:
I was just asking, it's fine if it stays and as you said it can be useful. Just wanted to point it out so that we don't have useless line of code